snapcraft/internal/repo/_deb.py

```diff
@@ -424,20 +427,28 @@ def install_stage_packages(
             ):
                 logger.debug(f"Extracting stage package: {pkg_name}")
                 installed.add(f"{pkg_name}={pkg_version}")
-                with tempfile.TemporaryDirectory(suffix="deb-extract") as extract_dir:
-                    # Extract deb package.
-                    cls._extract_deb(str(dl_path), extract_dir)
-
-                    # Mark source of files.
-                    marked_name = f"{pkg_name}={pkg_version}"
-                    cls._mark_origin_stage_package(extract_dir, marked_name)
-
-                    # Stage files to install_dir.
-                    file_utils.link_or_copy_tree(extract_dir, install_dir)
+                file_utils.link_or_copy(
+                    str(dl_path),
+                    (stage_packages_path / os.path.basename(str(dl_path))).as_posix(),
```

Comment:
dl_path.name gets the base name

Reply:
Also mixed use of str() and as_posix().  I'd suggest always using `str` unless there is a good reason for posix?

Reply:
I just copied that over, please make the code suggestion with the change if that is what you are suggesting :-)